src/coreclr/jit/compiler.cpp

```diff
@@ -5106,9 +5104,27 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
     compJitTelemetry.NotifyEndOfCompilation();
 #endif
 
-#if defined(DEBUG)
-    ++Compiler::jitTotalMethodCompiled;
-#endif // defined(DEBUG)
+    unsigned methodsCompiled = (unsigned)InterlockedIncrement((LONG*)&Compiler::jitTotalMethodCompiled);
+
+    if (JitConfig.DumpJittedMethods() && !compIsForInlining())
+    {
+#ifdef DEBUG
+        const int BUFSIZE            = 20;
+        char      osrBuffer[BUFSIZE] = {0};
+        if (opts.IsOSR())
+        {
+            // Tiering name already includes "OSR", we just want the IL offset
+            sprintf_s(osrBuffer, BUFSIZE, " @0x%x", info.compILEntry);
+        }
+
+        printf("%4d) JIT compiled [%s] %s   ILSize=%u, AsmSize=%u, hash=0x%08x %s%s\n", methodsCompiled,
+               compGetTieringName(), info.compFullName, info.compILCodeSize, *methodCodeSize, info.compMethodHash(),
+               osrBuffer, compGetStressMessage());
```

Comment:
Another option: put the compilation name first, then all the attributes following in `[` `]`:

```suggestion
        printf("%4d: JIT compiled %s [%s%s, IL size=%u, code size=%u, hash=0x%08x %s]\n", methodsCompiled,
               info.compFullName, compGetTieringName(), osrBuffer, info.compILCodeSize, *methodCodeSize, info.compMethodHash(),
               compGetStressMessage());
```

This makes more sense to me